Xamarin.PropertyEditing.Mac/Controls/TimeSpanEditorControl.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections;
+using AppKit;
+using Xamarin.PropertyEditing.Mac.Resources;
+using Xamarin.PropertyEditing.ViewModels;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	public class TimeSpanValidationTextDelegate : IValidationTextDelegate
+	{
+		public bool IsValid (NSText textObject) => TimeSpan.TryParse (textObject.Value, out var result);
+	}
+
+	internal class TimeSpanEditorControl : PropertyEditorControl<PropertyViewModel<TimeSpan>>
+	{
+		public TimeSpanEditorControl ()
+		{
+			StringEditor = new ValidationTextField (new TimeSpanValidationTextDelegate ()) {
+				BackgroundColor = NSColor.Clear,
+				ControlSize = NSControlSize.Small,
+				Font = NSFont.FromFontName (DefaultFontName, DefaultFontSize),
+				StringValue = string.Empty,
+				TranslatesAutoresizingMaskIntoConstraints = false,
+			};
+
+			// update the value on keypress
+			StringEditor.Changed += (sender, e) => {
+				if (TimeSpan.TryParse (StringEditor.StringValue, out TimeSpan result)) {
```

Comment:
Isn't that duplicating work with the time span validation above?

Reply:
Yeah.. I'm too defensive